source/extensions/filters/http/rate_limit_quota/filter.cc

```diff
@@ -0,0 +1,174 @@
+#include "source/extensions/filters/http/rate_limit_quota/filter.h"
+
+#include <memory>
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace RateLimitQuota {
+
+void RateLimitQuotaFilter::setDecoderFilterCallbacks(
+    Http::StreamDecoderFilterCallbacks& callbacks) {
+  callbacks_ = &callbacks;
+}
+
+// TODO(tyxia) Mostly are example/boilerplate code, polish implementation here.
+Http::FilterHeadersStatus RateLimitQuotaFilter::decodeHeaders(Http::RequestHeaderMap& headers,
+                                                              bool) {
+  // Start the stream on the first request.
+  auto start_stream = rate_limit_client_->startStream(callbacks_->streamInfo());
+  if (!start_stream.ok()) {
+    // TODO(tyxia) Consider adding the log.
+    return Envoy::Http::FilterHeadersStatus::Continue;
```

Comment:
Is this logic correct? Shouldn't it be subject to the no-assignment configuration?